src/utilities/ToolTipTimestamp.tsx

```diff
@@ -0,0 +1,35 @@
+import { Tooltip, Typography } from "@mui/material";
+import formatTimestamp from "./formatTimestamp";
+import formatToolTipTimestamp from "./formatToolTipTimestamp";
+
+type ToolTipTimestampProps = {
+  ToolTipTimestamp: string;
+};
+
+const styles = {
+  toolTipText: {
+    cursor: "pointer",
+    "&:hover": {
+      textDecoration: "underline",
+    },
+  },
+};
+const ToolTipTimestamp = ({ ToolTipTimestamp }: ToolTipTimestampProps) => {
+  return (
+    <Tooltip
+      title={formatToolTipTimestamp(ToolTipTimestamp)}
+      placement="bottom"
+      slotProps={{
+        popper: {
+          modifiers: [{ name: "offset", options: { offset: [5, -15] } }],
+        },
+      }}
+    >
+      <Typography variant="subtitle2" sx={styles.toolTipText}>
```

Comment:
Let's add a prop for `variant` and default it to `subtitle2`. In some cases we want to change this type, for example in `DirectMessage` it should be the `body2` variant. We can get the variant type from MUI's `TypographyVariant` type

So it'd look something like this:

```
type ToolTipTimestampProps = {
  ToolTipTimestamp: string;
  variant?: TypographyVariant;
};

...

const ToolTipTimestamp = ({ ToolTipTimestamp, variant = "subtitle2" }: ToolTipTimestampProps) => {
  return (
    <Tooltip
      placement="bottom"
      ...
    >
      <Typography variant={variant} sx={styles.toolTipText}>
        {formatTimestamp(ToolTipTimestamp)}
      </Typography>
    </Tooltip>
  );
};

```

Reply:
Thats a neat way to make it more dynamic. I have included the changes